### PR TITLE
bumped go version to 1.22

### PR DESCRIPTION
**Pull request by mzeevi**
This PR fixes some tests that were broken by API changes to container-app-operator and also addressed module and deps changes needed to bump Capp and bump Go to 1.22

### CODE REVIEW

**Comment by mzeevi** [comment]
/hold

Wait until some PRs in container-app-operator are merged

**Review by fran2810 — APPROVED** [review]
looks good i don't  understand why the secret change should be in this commit next time make it in separate commits

**Pull request merged**
